pkg/misc/depversion/depversion.go

```diff
@@ -188,7 +191,7 @@ func isSemver(s string) bool {
 }
 
 func isSemVerWildcard(s string) bool {
-	return !exactSvR.Match([]byte(s)) && exactSvRWithWildcard.Match([]byte(s))
+	return !exactSvR.Match([]byte(s)) && (exactSvRWithWildcardAndTidle.Match([]byte(s)) || exactSvRWithWildcard.Match([]byte(s)))
```

Comment:
Instead of using the regex you added, just first check if the string starts with tilde, and remove if found, then use existing `exactSvRWithWildcard`

Reply:
I have got the code to check for a tilde, and then remove the tilde. Because of this I have removed the regex. I have also combined the wildcard and tilde checks together because they both can start with a tilde.